src/TemplateCompiler/TypeAnalyzer/TemplateVariableTypesResolver.php

```diff
@@ -47,6 +49,23 @@ public function resolveArray(Array_ $array, Scope $scope): array
             $variableNamesToTypes[] = new VariableAndType($keyName, $variableType);
         }
 
+        if ($calledOnType) {
+            $calledOnReflection = $calledOnType->getClassReflection();
+            if ($calledOnReflection) {
+                $calledOnNativeReflection = $calledOnReflection->getNativeReflection();
+                if ($calledOnNativeReflection instanceof ReflectionClass) {
+                    foreach ($calledOnNativeReflection->getProperties() as $property) {
+                        if ($property->isPublic()) {
+                            $variableNamesToTypes[] = new VariableAndType(
+                                $property->getName(),
+                                $scope->getType($property)
```

Comment:
I tried to push this pull request a little further, but I am having trouble here. How can we get the properties of `$calledOnType` listed with name and their Type? I tried via reflection, but I don't see how I can go from https://apiref.phpstan.org/1.10.x/PHPStan.BetterReflection.Reflection.Adapter.ReflectionProperty.html to https://apiref.phpstan.org/1.10.x/PHPStan.Type.Type.html.

Any advice?